### PR TITLE
WIP - Closes #6 Built-in compiler

### LINKED ISSUE #6: Create builtin Elixir compiler to be able to include it in projects

### PR DESCRIPTION
**Pull request by wende**
Closes #6

### CODE REVIEW

File: elixir-stuff/elmchemy/lib/XBasics.ex [hunk 1]
```diff
@@ -7,62 +7,67 @@ defmodule Elmchemy.XBasics do
...
-  curry //2
-  curry div/2
-  curry rem/2
+     curry +/2
+     curry -/2
+     curry */2
+     curry //2
+     curry div/2
+     curry rem/2
+     # curry abs/2
```
Comment (wende) [inline elixir-stuff/elmchemy/lib/XBasics.ex]: Why is abs commented out?

File: elixir-stuff/elmchemy/lib/XBasics.ex [hunk 2]
```diff
@@ -7,62 +7,67 @@ defmodule Elmchemy.XBasics do
...
-    if a > b do
-      :g_t
-    else if a < b do
-      :l_t
-    else
-      :e_q
+    cond do
+      a > b -> :g_t
+      a < b -> :l_t
+      true -> :e_q
```
Comment (wende) [inline elixir-stuff/elmchemy/lib/XBasics.ex]: Nice

File: elmchemy [hunk 3]
```diff
@@ -30,23 +37,24 @@ case "$1" in
...
-        node elmchemy.js .elmchemy_output > .elmchemy_elixir_output
+        node $SOURCE_DIR/elmchemy.js .elmchemy_output > .elmchemy_elixir_output
         current_file=""
         while IFS= read -r line; do
             if [[ $line =~ ">>>>" ]]; then
                 current_file="${line/\/\///}"
                 current_file="${current_file%%.*}.ex"
-                current_file="${current_file/>>>>/.elmchemy/}"
-                echo $current_file
+                current_file="${current_file/>>>>$2/$3}"
```
Comment (wende) [inline elmchemy]: What's $3 in this context?

File: src/ExExpression.elm [hunk 4]
```diff
@@ -475,10 +491,14 @@ operators =
...
-    , ( "rem", "" ) -- Exception
-    , ( "^", "" ) -- Exception
+    , ( "%", "" )
+      -- Exception
+    , ( "//", "" )
+      -- Exception
+    , ( "rem", "" )
+      -- Exception
+    , ( "^", "" )
+      -- Exception
```
Comment (wende) [inline src/ExExpression.elm]: Should move the comments above, not below. Not sure if it's elm format or sth else

File: elixir-stuff/elmchemy/lib/XBasics.ex (hunk 1 above)
Comment (Baransu) [inline elixir-stuff/elmchemy/lib/XBasics.ex]: It was causing Elixir compilation fail

File: elmchemy (hunk 3 above)
Comment (Baransu) [inline elmchemy]: It's output directory

File: src/ExExpression.elm (hunk 4 above)
Comment (Baransu) [inline src/ExExpression.elm]: We probably have different `elm-format` version. We can set [lint-staged](https://github.com/okonet/lint-staged) for that so formatting will be consistent.

File: elmchemy (hunk 3 above)
Comment (Baransu) [inline elmchemy]: `$1` - aggregate
`$2` - src dir
`$3` - output dir

**Pull request merged**